docs_src/design/adr/0019-Service-List.md

```diff
@@ -0,0 +1,87 @@
+- The service list would have to be a list of services and service locations (URI).  Therefore, the service list will really be a map of service names (keys) and the location of the services URIs (values).  This will help support things like the API gateway setup as well.  In some cases, just a list of the service names is needed (in security token setup for example).  In this latter case, the service would just pick off the service keys.
+- Generation of security tokens for the new services under most of the conceived alternatives is complex.  In the current security bootstrapping, secret store setup will require a restart of all the services in order to add or remove a service.  The ORRA project and some adopters are clear that they would like to have the ability to add/remove services dynamically at runtime.
+- From a security perspective, being able to add/remove new services dynamically opens up a rather large attack vector and vulnerability.
+
+### Implementation Alternative Discussion
+
+- A service list could be made available in a configuration file in the file system (non-Docker) or volume mount (in a Docker situation).
+  - All services would get the configuration in a consistent way (#1)
+  - It would also mean that there is a single, authoritative list for all services and it could be made to be ACID safe (#2 and #3).
+  - The file would have to be validated whenever edited in order not to bring down all the services with bad contents.
+  - This alternative would solve the requirement that the solution operate with or without Consul (#4).  UI and CLI should have access to the same file or volume mount.
+  - There is a question about who or what service would put this file into the file system or volume mount – especially since security bootstrapping would need it first but this file would be needed in non-secure runtime as well.
+  - There isn’t a central “dummy” service that would own the file – especially for dev/test.
+  - It has to be a service that is always there (like core metadata), but also available from the start of all services (which metadata is not in the case of security services).
+  - Services would have to monitor the file or volume for changes in order to be able to address the add/removal of services at runtime.
+  - How would the security bootstrapping address the add/removal of services and get new tokens established?  The bootstrapping service isn’t even long running to do this.
+- Have global configuration in Consul (or config/reg service) that everyone reads.
+  - When Consul is not used, have a common service list in each service configuration.toml file that everyone reads.
+  - While the means to get the configuration is consistent in this solution alternative (#1), there is not a single, authoritative list of services when using local configuration and making ACID transactions (#3) across the multiple files would be difficult – especially if distributed across hosts.
```

Comment:
This bullet item is confusing.  This section is supposed to be talking about "Have global configuration in Consul (or config/reg service) that everyone reads."  Also, aren't EdgeX's configuration.toml considered read-only?

Reply:
tried to clarify and clean this up.  Bottom line, Consul could be used to make things consistent and ACID safe with regard to service list, but since people can use configuration files in place of Consul.  When each service has its own config file (and doesn't use Consul), consistency and any ideas of ACID transactions against the multiple configuration files goes bye bye.